frontend/src/components/SimpleCarousel/SimpleCarousel.js

```diff
@@ -0,0 +1,127 @@
+import ArrowRight from '@mui/icons-material/ArrowForwardIosOutlined'
+import Box from '@mui/material/Box'
+import Grid from '@mui/material/Grid'
+import IconButton from '@mui/material/IconButton'
+import PropTypes from 'prop-types'
+
+import { getOuterWidth } from './SimpleCarousel.utils'
+
+export const SimpleCarouselItem = forwardRef(({ children, sx }, ref) => {
+  return (
+    <Box sx={{ ...sx, display: 'inline-block' }} ref={ref}>
+      {children}
+    </Box>
+  )
+})
+
+export const SimpleCarousel = ({ children, value }) => {
+  const itemRef = useRef([])
+  const containerRef = useRef(null)
+  const innerRef = useRef(null)
+  const childrenArray = useMemo(() => (Array.isArray(children) ? children : [children]), [children])
+  const [scrollPos, setScrollPos] = useState(0)
+  const [containerHeight, setContainerHeight] = useState(0)
+  const [scrollSize, setScrollSize] = useState(0)
+  const showArrows = scrollSize > 0
+
+  const handleLeft = useCallback(() => {
+    setScrollPos(scrollPos => {
+      const idx = itemRef.current.findIndex(el => {
+        return el.offsetLeft >= scrollPos
+      })
+      const newPos = idx === 0 ? 0 : itemRef.current[idx - 1].offsetLeft
+      return Math.max(newPos, 0)
+    })
+  }, [])
+
+  const handleRight = useCallback(() => {
+    setScrollPos(scrollPos => {
+      const idx = itemRef.current.findIndex(el => {
+        return el.offsetLeft <= scrollPos && el.offsetLeft + getOuterWidth(el) > scrollPos
+      })
+      const newPos = idx + 1 === itemRef.current.length ? scrollSize : itemRef.current[idx + 1].offsetLeft
+      return Math.min(newPos, scrollSize)
+    })
+  }, [scrollSize])
+
+  useEffect(() => {
+    const setDimensions = () => {
+      setContainerHeight(innerRef.current.clientHeight)
+      setScrollSize(Math.max(0, innerRef.current?.clientWidth - containerRef.current?.clientWidth))
+    }
+    setDimensions()
+    window.addEventListener('resize', setDimensions)
+    return () => window.removeEventListener('resize', setDimensions)
+  }, [showArrows])
+
+  useEffect(() => {
+    innerRef.current.style.left = `${-scrollPos}px`
+  }, [scrollPos])
+
+  useEffect(() => {
+    const containerWidth = containerRef.current.clientWidth
+    const scrollPos = -innerRef.current.offsetLeft
+    const valueIndex = childrenArray.findIndex(child => child.props.value === value)
+    const item = itemRef.current[valueIndex]
+    if (!item) {
+      return
+    }
+    if (item.offsetLeft - scrollPos < 0) {
+      setScrollPos(item.offsetLeft)
+    } else if (item.offsetLeft + item.clientWidth - scrollPos > containerWidth) {
+      setScrollPos(item.offsetLeft + item.clientWidth - containerWidth)
+    }
+  }, [childrenArray, value])
+
+  return (
+    <Grid container alignItems="center">
+      {showArrows && (
+        <Grid item>
+          <IconButton onClick={handleLeft} disabled={scrollPos === 0}>
+            <ArrowLeft fontSize="small" />
+          </IconButton>
+        </Grid>
+      )}
```

Comment:
Could you make the left and right <> buttons disappear when disabled?

Reply:
I also tried to do so, but if we make them disappear, the container width grows by the 2 arrow widths, and it affects the scroll effect. I tried to make them have absolute position and that requires to have white background for the arrows, but the problem is the header background has opacity value, and adding another half-transparent opacity will make it solid white background.

Reply:
I'm ok with making the background opaque to make this easier. Absolute positioning over the ends of the scroll container would simplify things.